docs/architecture.md

```diff
@@ -0,0 +1,127 @@
+---
+hide_table_of_contents: false
+title: DBOS Architecture
+---
+
+## DBOS Architecture
+
+DBOS provides a lightweight library for durable workflows built on top of Postgres.
+Architecturally, an application built with DBOS looks like this:
+
+<img src={require('@site/static/img/architecture/dbos-architecture.png').default} alt="DBOS Architecture" width="750" className="custom-img"/>
+
+You install the DBOS library into your existing application running on your existing servers and infrastructure and connect it to a Postgres database.
+Then, you annotate workflows and steps in your application code.
+The DBOS library **checkpoints** those workflows and steps in Postgres.
+If your program fails, crashes, or is interrupted, DBOS uses those checkpoints to automatically recover your workflows from the last completed step.
+
+<img src={require('@site/static/img/architecture/dbos-steps.jpg').default} alt="DBOS Steps" width="750" className="custom-img"/>
+
+## Comparison to External Workflow Orchestrators
```

Comment:
This feels redundant with https://docs.dbos.dev/why-dbos#dbos-vs-other-systems 

Consider shortening or linking / combining?